File: frontend/lib/models/document.dart
class Document {
  final int id;
  final String titre;
  final String? description;
  final String type;
  final String filePath;
  final String fileName;
  final String? mimeType;
  final int? fileSize;

  Document({
    required this.id,
    required this.titre,
    this.description,
    this.type = 'autre',
    required this.filePath,
    required this.fileName,
    this.mimeType,
    this.fileSize,
  });

  factory Document.fromJson(Map<String, dynamic> json) {
    return Document(
      id: json['id'],
      titre: json['titre'],
      description: json['description'],
      type: json['type'] ?? 'autre',
      filePath: json['file_path'] ?? '',
      fileName: json['file_name'] ?? '',
      mimeType: json['mime_type'],
      fileSize: json['file_size'],
    );
  }
}

File: backend/temp_backend/app/Models/Document.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Facades\Storage;

class Document extends Model
{
    use HasFactory;

    protected $fillable = [
        'titre',
        'description',
        'type',
        'file_path',
        'file_name',
        'mime_type',
        'file_size',
        'created_by',
    ];

    public function createdBy()
    {
        return $this->belongsTo(User::class, 'created_by');
    }

    public function getUrlAttribute(): ?string
    {
        return $this->file_path ? Storage::url($this->file_path) : null;
    }
}
